src/utils/gitUtils.ts

```diff
@@ -0,0 +1,121 @@
+
+    if (repo) {
+        const originUrl: string | undefined = await tryGetRemote(uri.fsPath);
+        gitWorkspaceState.repo = repo;
+        gitWorkspaceState.dirty = !!repo.state.workingTreeChanges.length;
+        gitWorkspaceState.remoteUrl = originUrl;
+        gitWorkspaceState.hasAdminAccess = originUrl ? hasAdminAccessToRepo(await (tryGetReposGetResponseData(context, originUrl))) : false;
+    }
+
+    return gitWorkspaceState;
+}
+
+export async function verifyRepoForCreation(context: IActionContext, gitWorkspaceState: GitWorkspaceState, uri: Uri): Promise<Repository> {
+    if (!gitWorkspaceState.repo) {
+        const gitRequired: string = localize('gitRequired', 'A GitHub repository is required to create a Static Web App. Would you like to initialize your project as a git repository and create a GitHub remote?');
+        context.telemetry.properties.cancelStep = 'initRepo';
+
+        await ext.ui.showWarningMessage(gitRequired, { modal: true }, DialogResponses.yes);
+        const gitApi: API = await getGitApi();
+        const newRepo: Repository | null = await gitApi.init(uri);
+        if (!newRepo) {
+            throw new Error(localize('gitInitFailed', 'Git initialization failed.  Please initialize a git repository manually and attempt to create again.'));
+        }
+
+        await promptForCommit(newRepo, localize('initCommit', 'Initial commit'));
+        gitWorkspaceState.repo = newRepo;
+
+    } else if (!!gitWorkspaceState.remoteUrl && !gitWorkspaceState.hasAdminAccess) {
+        context.telemetry.properties.cancelStep = 'adminAccess';
+
+        const adminAccess: string = localize('adminAccess', 'Admin access to the GitHub repository is required.  Please use a repo with admin access or create a fork.');
+        await ext.ui.showWarningMessage(adminAccess, { modal: true });
+
+    } else if (gitWorkspaceState.dirty && gitWorkspaceState.repo) {
+        context.telemetry.properties.cancelStep = 'dirtyWorkspace';
+
+        const commitChanges: string = localize('commitChanges', 'Commit all working changes to create a Static Web App.');
+        await ext.ui.showWarningMessage(commitChanges, { modal: true }, { title: localize('commit', 'Commit') });
+        await promptForCommit(gitWorkspaceState.repo);
+        gitWorkspaceState.dirty = false;
+    }
+
+    return gitWorkspaceState.repo;
+}
+
+export async function promptForCommit(repo: Repository, value?: string): Promise<void> {
+    const commitPrompt: string = localize('commitPrompt', 'Enter a commit message.');
+    // VS Code doesn't handle nested git repos very well; it'll show changes as working, but is unable to commit
+    // so commit with { empty: true } so that we don't get errors when we making "empty" commits
+    const commitOptions: CommitOptions = { all: true, empty: true };
+
+    const commitMsg: string = await ext.ui.showInputBox({ prompt: commitPrompt, placeHolder: `${commitPrompt}..`, value });
```

Comment:
I think it looks kinda redundant to have `commitPrompt` for both the prompt and placeholder. Maybe do something like `value: value || 'Add local changes'` instead of a placeholder?

![Screen Shot 2021-04-08 at 3 33 44 PM](https://user-images.githubusercontent.com/22795803/114104616-0f02a700-9880-11eb-8016-2a1fc01115d1.png)
![Screen Shot 2021-04-08 at 3 37 24 PM](https://user-images.githubusercontent.com/22795803/114104752-5426d900-9880-11eb-8c73-cb38bd56b5f4.png)

Reply:
Hmnm, I agree that having the prompt and placeholder being the same looks kind of redundant, but I also prefer just having no placeholder then (and force the user to write something) than just a generic "Add local changes".  If we do want to have a default commit message, I'd prefer something more like "Commit made by Azure Static Web Apps extension" or something that the user will know exactly where the commit came from.

Reply:
I think "Commit made by Azure Static Web Apps extension" would be good. Just having something as a placeholder is nice. My main concern is it can be hard to come up with a commit message on the spot